src/transformers/models/fsmt/modeling_fsmt.py

```diff
@@ -276,373 +276,699 @@
 """
 
 
-def invert_mask(attention_mask):
-    """Turns 1->0, 0->1, False->True, True-> False"""
-    assert attention_mask.dim() == 2
-    return attention_mask.eq(0)
+# Copied from transformers.models.mbart.modeling_mbart.shift_tokens_right
+def shift_tokens_right(input_ids, pad_token_id):
```

Comment:
I'm actually not sure whether using MBart's way of creating the tokens here is correct. Note that MBart uses a very special language-dependent pattern when fine-tuning so that the `decoder_start_token_id` depends on the language, see Fig. 1 of [paper](https://arxiv.org/pdf/2001.08210.pdf). 

IMO, this is not the case for FSMT. FSMT always uses the EOS/SEP token as the `decoder_start_token_id = sep_token_id = "<\s>"` if I understand correctly => Think it's better therefore to copy the `shift_tokens_right` function from Bart and set `decoder_start_token_id = 2` as a default in the config. This would make this function much easier. 

Would be great if you could confirm @stas00

Reply:
Yes, you are right. This is also equivalent in that, inf FSMT sequence ends with `EOS` and then this method moves it to the beginning. But yeah, I will copy  BARTs  function, which is more explicit here.

Reply:
Indeed! Thank you for catching that, @patrickvonplaten!